app/lib/bk/compat/parsers/circleci/orbs.rb

```diff
@@ -28,6 +19,53 @@ def translator(action, _config)
           ]
         end
       end
+
+      # Docker orb specific implementation
+      class DockerOrb
+        def matcher(orb, _conf)
+          orb.start_with?('docker/')
+        end
+
+        def translator(action, config)
+          method_map = {
+            'docker/build' => -> { translate_docker_build(config) },
+            'docker/install-docker' => -> { validate_and_translate_install_docker(config) }
+          }
+          method_map.fetch(action, -> { ["# Unsupported docker orb action: #{action}"] }).call
```

Comment:
very nice pattern but has the disadvantage of calling both translations just to create the mapping in addition to creating unnecessary pass-through lambdas. Instead I would do something like the following

```suggestion
          method_map = {
            'docker/build' => 'translate_docker_build',
            'docker/install-docker' => 'validate_and_translate_install_docker'
          }
          send(method_map.fetch(action, 'default_translation'), config)
```

and create a `default_translation` method in the class that returns the array.

If you rename the methods you can even do something like what is done [in the harness step translator](https://github.com/buildkite/migration/blob/819da4380308094ea7bb1caf31d80ac2b7ad8370/app/lib/bk/compat/parsers/harness/steps.rb#L19), [in the circleci executor translator](https://github.com/buildkite/migration/blob/819da4380308094ea7bb1caf31d80ac2b7ad8370/app/lib/bk/compat/parsers/circleci/executors.rb#L43) and [gha action translator](https://github.com/buildkite/migration/blob/819da4380308094ea7bb1caf31d80ac2b7ad8370/app/lib/bk/compat/parsers/gha/steps/actions.rb#L17)